core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -373,4 +434,196 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
+        }
+        if (fieldsBoosts == null || fieldsBoosts.isEmpty()) {
+            validationException = addValidationError("no fields specified for multi_match query.", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    protected MultiMatchQueryBuilder doReadFrom(StreamInput in) throws IOException {
+        MultiMatchQueryBuilder multiMatchQuery = new MultiMatchQueryBuilder(in.readGenericValue(),
+                (Map<String, Float>) in.readGenericValue());
+        if (in.readBoolean()) {
+            multiMatchQuery.type = Type.valueOf(in.readString());
+        }
+        if (in.readBoolean()) {
+            multiMatchQuery.operator = Operator.readOperatorFrom(in);
+        }
+        multiMatchQuery.analyzer = in.readOptionalString();
+        multiMatchQuery.slop = in.readOptionalVInt();
+        if (in.readBoolean()) {
+            multiMatchQuery.fuzziness = Fuzziness.readFuzzinessFrom(in);
+        }
+        multiMatchQuery.prefixLength = in.readOptionalVInt();
+        multiMatchQuery.maxExpansions = in.readOptionalVInt();
+        multiMatchQuery.minimumShouldMatch = in.readOptionalString();
+        multiMatchQuery.fuzzyRewrite = in.readOptionalString();
+        multiMatchQuery.useDisMax = in.readOptionalBoolean();
+        multiMatchQuery.tieBreaker = in.readOptionalFloat();
+        multiMatchQuery.lenient = in.readOptionalBoolean();
+        multiMatchQuery.cutoffFrequency = in.readOptionalFloat();
+        if (in.readBoolean()) {
+            multiMatchQuery.zeroTermsQuery = ZeroTermsQuery.valueOf(in.readString());
+        }
+        return multiMatchQuery;
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        out.writeGenericValue(value);
+        out.writeGenericValue(fieldsBoosts);
+        if (type != null) {
+            out.writeBoolean(true);
+            out.writeString(type.name().toUpperCase(Locale.ROOT));
+        } else {
+            out.writeBoolean(false);
+        }
+        if (operator != null) {
+            out.writeBoolean(true);
+            operator.writeTo(out);
+        } else {
+            out.writeBoolean(false);
+        }
+        out.writeOptionalString(analyzer);
+        out.writeOptionalVInt(slop);
+        if (fuzziness != null) {
+            out.writeBoolean(true);
+            fuzziness.writeTo(out);
+        } else {
+            out.writeBoolean(false);
+        }
+        out.writeOptionalVInt(prefixLength);
+        out.writeOptionalVInt(maxExpansions);
+        out.writeOptionalString(minimumShouldMatch);
+        out.writeOptionalString(fuzzyRewrite);
+        out.writeOptionalBoolean(useDisMax);
+        out.writeOptionalFloat(tieBreaker);
+        out.writeOptionalBoolean(lenient);
+        out.writeOptionalFloat(cutoffFrequency);
+        if (zeroTermsQuery != null) {
```

Comment:
fyi, I added serialization to the enum I moved to MatchQuery (#13402) maybe we can reuse this.